src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -447,6 +460,85 @@ func (r *StatefulSetResource) queryRedpandaStatus(
 	return nil
 }
 
+// Temporarily using the status/ready endpoint until we have a specific one for restarting.
+func (r *StatefulSetResource) queryRedpandaUnderReplicatedPartition(
+	ctx context.Context, adminURL *url.URL,
+) error {
+	client := &http.Client{Timeout: adminAPITimeout}
+
+	// TODO right now we support TLS only on one listener so if external
+	// connectivity is enabled, TLS is enabled only on external listener. This
```

Comment:
I don't think this is true, tls could be enabled on internal listener as well. I think the behavior you're describing is true only for schema registry.

Reply:
To be brutally honest I copied it from your previous implementation 

https://github.com/redpanda-data/redpanda/blob/a6a021bc562f692eaf1793fcc8a6bb179318e374/src/go/k8s/pkg/resources/statefulset_update.go#L417-L419

https://github.com/redpanda-data/redpanda/commit/6235e96ee7b0b2bd9d489d9a3ae26cde8786abff#diff-5619b1cd9bdb790f53d380d84b884e952da245749cb78145fdf8f6b328b1be2cR182-R184